src/test/java/org/commcare/formplayer/tests/SnapshotTests.java

```diff
@@ -0,0 +1,110 @@
+package org.commcare.formplayer.tests;
+
+import org.apache.commons.io.FileUtils;
+import org.commcare.formplayer.sandbox.UserSqlSandbox;
+import org.commcare.formplayer.sqlitedb.UserDB;
+import org.commcare.formplayer.utils.GenerateSnapshotDatabases;
+import org.commcare.formplayer.utils.TestContext;
+import org.javarosa.core.services.storage.IMetaData;
+import org.javarosa.core.services.storage.IStorageUtilityIndexed;
+import org.javarosa.core.util.Iterator;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Tests the compatibility of the current code runtime against the currently captured database
+ * snapshots.
+ *
+ * If these tests fail, the snapshots need to be rebuild and databases will need to be cold started
+ * in any existing environments after updating.
+ *
+ * @author ctsims
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = TestContext.class)
+public class SnapshotTests extends BaseTestClass {
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        copySnapshotResources();
+    }
+
+    private void copySnapshotResources() {
+        try {
+            File destDirectory = new File(getDatabaseFolderRoot());
+
+            FileUtils.copyDirectory(new File(GenerateSnapshotDatabases.snapshotDbDirectory)
+                    , destDirectory);
+        } catch (IOException exception) {
+            exception.printStackTrace();
+            Assert.fail("Error moving snapshot data to test directory");
+        }
+    }
+
+    protected boolean remoteDatabaseFoldersAfterTests() {
+        return true;
+    }
+
+
+    @Test
+    public void testUserSandbox() throws Exception {
+        UserDB database = getUserDbConnector("snapshot", "snapshot_test", null);
+        if (!database.databaseFileExists()) {
+            Assert.fail("Snapshot UserDB Missing for tests you may need to rebuild the snapshot " +
+                    "with the CreateSnapshotDbs Gradle Task ");
+        }
+        //Try to enumerate records of each type
+        UserSqlSandbox sandbox = new UserSqlSandbox(database);
+        enumerate("Case", sandbox.getCaseStorage());
+        enumerate("Ledger", sandbox.getLedgerStorage());
+        enumerate("User",sandbox.getUserStorage());
```

Comment:
Very small nit, but spacing in this block is inconsistent.

Reply:
thanks! Fixed.